stampifier.py

```diff
@@ -1,41 +1,65 @@
-''' This module contains the main interface definition
-for the classifier and summarizer module
-'''
+"""This script creates Stampify"""
+
+import logging
+
 from classification.classifier_main import Classifier
+from data_models.website import Website
+from extraction import extractor
+from stamp_generation.stamp_generator import StampGenerator
 from summarization.extractor_output_preprocessor import \
     ExtractorOutputPreprocessor
 from summarization.summarizer_main import Summarizer
 
+LOGGER = logging.getLogger()
+LOG_FILENAME = 'website.log'
+logging.basicConfig(filename=LOG_FILENAME, level=logging.DEBUG)
 
-class ClassifierAndSummarizer:
-    def __init__(self, contents, max_pages):
-        '''
-        Params:
-          * contents : an object of type contents
-            it contains the contents_list
-          * max_pages : the maximum pages required
-        '''
-        self.contents = contents
+
+class Stampifier:
+
+    def __init__(self, url, max_pages):
+        self.url = url
         self.max_pages = max_pages
+        self._website = Website(self.url)
         self.is_stampifiable = False
         # object of type StampPages
         self.stampified_pages = None
 
-    def _preprocess_contents(self):
-        '''
-        This method will use ExtractorOutputPreprocessor
-        to split the contents into different types
-        '''
-        output_preprocessor = ExtractorOutputPreprocessor(self.contents)
-        self.preprocessed_contents_dict \
-            = output_preprocessor.get_preprocessed_content_lists()
-        self.title_text_contents \
-            = self.preprocessed_contents_dict["titles"]
-        self.normal_text_contents \
-            = self.preprocessed_contents_dict["sentences"]
-        self.media_contents = self.preprocessed_contents_dict["media"]
-        self.embedded_contents \
-            = self.preprocessed_contents_dict["embedded_content"]
+    def stampify(self):
+        """Starts the stampification process"""
+
+        if not self._website.is_valid:
+            LOGGER.error("Invalid URL!")
```

Comment:
Isn't it better to throw exception instead of returning?

Reply:
Yes, It is. Thanks for suggestion! Updated accordingly. Let me know if any changes required in that!